From 50b2e0bd6e74d40bdfaa8bf2edf860c20393aa68 Mon Sep 17 00:00:00 2001
From: Gwendal Grignou <gwendal@chromium.org>
Date: Thu, 18 Dec 2014 16:36:00 -0800
Subject: [PATCH 2/7] ec util: do not compile IO mapped memory related code for
 mips.

MIPS host CPU does not access IO registers using a separate address space.
Remove for support of LPC for this host architecture.

Confine x86 function to Intel Architecture only.

BRANCH=none
TEST=Compile: Test on ARM, MIPS and X86: using emerge... ec-utils.
BUG=chromium:443783

Change-Id: I9d4276ec3588037adfcff96e596bbe8be74f22fd
Signed-off-by: Gwendal Grignou <gwendal@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/236687
Reviewed-by: Vic Yang <victoryang@chromium.org>
Reviewed-by: Randall Spangler <rspangler@chromium.org>
---
 util/ectool.c | 36 ++++++++++++++++++++++++------------
 1 file changed, 24 insertions(+), 12 deletions(-)

diff --git a/util/ectool.c b/util/ectool.c
index 7e5050d90..7bb2f2e2a 100644
--- a/util/ectool.c
+++ b/util/ectool.c
@@ -9,7 +9,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/io.h>
 #include <unistd.h>
 
 #include "battery.h"
@@ -756,6 +755,8 @@ int cmd_flash_protect(int argc, char *argv[])
 	return 0;
 }
 
+#ifdef __x86_64
+#include <sys/io.h>
 
 int cmd_serial_test(int argc, char *argv[])
 {
@@ -777,6 +778,28 @@ int cmd_serial_test(int argc, char *argv[])
 }
 
 
+int cmd_port_80_flood(int argc, char *argv[])
+{
+	int i;
+
+	for (i = 0; i < 256; i++)
+		outb(i, 0x80);
+	return 0;
+}
+#else
+int cmd_serial_test(int argc, char *argv[])
+{
+	printf("x86 specific command\n");
+	return -1;
+}
+
+int cmd_port_80_flood(int argc, char *argv[])
+{
+	printf("x86 specific command\n");
+	return -1;
+}
+#endif
+
 int read_mapped_temperature(int id)
 {
 	int rv;
@@ -3609,17 +3632,6 @@ int cmd_console(int argc, char *argv[])
 	printf("\n");
 	return 0;
 }
-
-/* Flood port 80 with byte writes */
-int cmd_port_80_flood(int argc, char *argv[])
-{
-	int i;
-
-	for (i = 0; i < 256; i++)
-		outb(i, 0x80);
-	return 0;
-}
-
 struct param_info {
 	const char *name;	/* name of this parameter */
 	const char *help;	/* help message */
-- 
2.18.0

